technique/reflectometry/base.py

```diff
@@ -0,0 +1,398 @@
+        print("Theta set to: {}".format(theta))
+        if not self.dry_run:
+            g.cset("theta", theta)
+
+    def _update_title(self, title):
+        if self.dry_run:
+            print("New Title: {}".format(title))
+            g.change_title(title)
+
+    def _set_height_offset(self, height):
+        print("Sample: height offset from beam={}".format(height))
+        if not self.dry_run:
+            g.cset("HEIGHT_OFFSET", height)
+
+    def _set_height2_offset(self, height, constants):
+        if constants.has_height2:
+            print("Sample: height2 offset from beam={}".format(height))
+            if not self.dry_run:
+                g.cset("HEIGHT2_OFFSET", height)
+        elif height != 0:
+            print("ERROR: Height 2 off set is being ignored")
+
+    def _set_translation(self, translation):
+        print("Translation to {}".format(translation))
+        if not self.dry_run:
+            g.cset("TRANSLATION", translation)
+            g.waitfor_move()
+
+    def _set_slit_gaps(self, theta, constants, s1vg, s2vg, s3vg, s4vg, sample):
+        """
+        Set the slit gaps either to user settings or calculated based on foot print and max slit size
+        Args:
+            theta: angle theta is set to
+            constants: machine constants
+            s1vg: s1 vertical gap set by user; None use footprint calculated gap
+            s2vg: s2 vertical gap set by user; None use footprint calculated gap
+            s3vg: s3 vertical gap set by user; None use percentage of maximum
+            s4vg: s4 vertical gap set by user; None use percentage of maximum
+            sample: sample parameters
+        """
+        s1, s2 = self._calculate_slit_gaps(theta, sample.footprint, sample.resolution, constants)
+
+        factor = theta / constants.maxTheta
+        s3 = constants.s3max * factor
+        s4 = constants.s4max * factor
+
+        if s1vg is not None:
+            s1 = s1vg
+
+        if s2vg is not None:
+            s2 = s2vg
+
+        if s3vg is not None:
+            s3 = s3vg
+
+        if s4vg is not None:
+            s4 = s4vg
+
+        print("Slit gaps 1-4 set to: {}, {}, {}, {}".format(s1, s2, s3, s4))
+        if not self.dry_run:
+            g.cset("S1VG", s1)
+            g.cset("S2VG", s2)
+            g.cset("S3VG", s3)
+            g.cset("S4VG", s4)
+
+    def _calculate_slit_gaps(self, theta, footprint, resolution, constants):
+        s1sa = constants.s1s2 + constants.s2sa
+        footprint_at_theta = footprint * sin(radians(theta))
+        s1 = 2 * s1sa * tan(radians(resolution * theta)) - footprint_at_theta
+        s2 = (constants.s1s2 * (footprint_at_theta + s1) / s1sa) - s1
+        return s1, s2
+
+    def _set_h_gaps(self, s1hg, s2hg, s3hg, s4hg):
+        print("Setting hgaps to {} (None's are not changed)".format([s1hg, s2hg, s3hg, s4hg]))
+        if not self.dry_run:
+            if s1hg is not None:
+                g.cset("S1HG", s1hg)
```

Comment:
Consider adding `dryrun` to cset is some way. Discussion is needed maybe make a ticket.